### ISSUE TITLE
Using a pre-signed URL to upload a file

### ISSUE THREAD

**Issue by trsteel88** [aws-amplify/amplify-js#1687]
** Which Category is your question related to? **

Uploading files with a pre-signed url (without cognito)

** What AWS Services are you utilizing? **

Amazon S3 Bucket Only

** Provide additional details e.g. code snippets **

Apologies if this is not the correct place to ask this question.

I have an API which already provides a pre-signed URL which will allow me to post a file to an Amazon S3 bucket. This pre-signed url designates the storage path and a specific Content MD5 hash which is allowed for the file.

Is there a way this package can be used for handling the upload to that pre-signed url? I am looking to handle this in React Native and Web. In React Native I want to be able to resume uploads if the user closes the app and re-opens it later.

Thank you.

**Comment by nidsharm** [comment]
Hi @trsteel88 ,

We do not support uploading an object using a presigned url in Amplify yet. You could however use the AWS js sdk [S3 getSignedURL API](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getSignedUrl-property) to upload the object.

**Comment by jordanranz** [comment]
It seems the question was answered here. Please reopen or open a new issue if there are anymore related questions or concerns.

**Comment by osama-rizk** [comment]
Hey everyone, 

 This feature has been released in `aws-amplify@6.16.4`
  (https://github.com/aws-amplify/amplify-js/releases/tag/aws-amplify%406.16.4) (via @aws-amplify/storage@6.14.0).
  You can now generate presigned URLs for uploads using the getUrl API with method: 'PUT':

```
  import { getUrl } from 'aws-amplify/storage';

  const { url } = await getUrl({
    path: 'uploads/file.pdf',
    options: {
      method: 'PUT',
      contentType: 'application/pdf',
      expiresIn: 3600,
    }
  });

  // Use with any HTTP client or custom library
  await fetch(url, {
    method: 'PUT',
    body: file,
    headers: { 'Content-Type': 'application/pdf' },
  });

```

  To upgrade: npm install aws-amplify@latest

Here is the docs link https://docs.amplify.aws/vue/frontend/storage/upload-files/#upload-using-a-presigned-url-1
